### PR TITLE
Metis docker + uwsgi support, and some performance tweaks for Jia + Metis dockers

### PR DESCRIPTION
**Pull request by marcua**

### CODE REVIEW

File: metis/metis/__init__.py [hunk 1]
```diff
@@ -1,3 +1,5 @@
+from gevent import monkey; monkey.patch_all()
```
Comment (usmanm) [inline metis/metis/__init__.py]: Does PySpark work with gevent?

File: jia/docker/supervisord.conf [hunk 2]
```diff
@@ -3,7 +3,7 @@ nodaemon=true
 childlogdir=/var/log/jia
 
 [program:jia]
-command=uwsgi --pythonpath /chronology/jia --socket 127.0.0.1:8152 --processes 4 --gevent 100 --module jia_uwsgi --callable app --logto /var/log/jia/jia.log
+command=uwsgi --pythonpath /chronology/jia --socket 127.0.0.1:8152 --processes 4 --gevent 100 --harakiri 300 --module jia_uwsgi --callable app --logto /var/log/jia/jia.log
```
Comment (usmanm) [inline jia/docker/supervisord.conf]: Does `harakiri` mean kill yo self after 300s?

File: metis/metis/__init__.py (hunk 1 above)
Comment (marcua) [inline metis/metis/__init__.py]: that's a good q.  i've been using the python executor, so it's hard for me to test :(

presumably we need to upgrade pyspark anyway for 1.3, so should we test it at that point, or should we test now?

File: jia/docker/supervisord.conf (hunk 2 above)
Comment (marcua) [inline jia/docker/supervisord.conf]: yea terrible name, but nginx/uwsgi were killing requests after 60s, so i wanted longer-running computations to be able to run.  5 minutes seemed reasonable...

File: metis/metis/__init__.py (hunk 1 above)
Comment (usmanm) [inline metis/metis/__init__.py]: Yeah, I'd imagine we should make sure it works with Spark.

File: metis/docker/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+RUN \
+  apt-get update && \
+  apt-get install -y software-properties-common python2.7 \
+  python-setuptools python-dev libevent-dev \
+  supervisor python-pandas nginx
+COPY chronology /chronology
+WORKDIR chronology/metis
+RUN easy_install pip
+RUN pip install -r requirements.txt
```
Comment (lydiagu) [inline metis/docker/Dockerfile]: one suggestion I read in the docker best practices is to explicitly copy over requirements.txt first, then copy the rest of the code. This way, if some other part of the code changed, you don't have to rerun pip install - requirements.txt

Comment (marcua) [inline metis/docker/Dockerfile]: sweet---fixed Jia and Metis to include this perf improvement.

File: metis/metis/__init__.py (hunk 1 above)
Comment (marcua) [inline metis/metis/__init__.py]: just ran tests, and bad news: previous commits have broken the spark executor.  i fixed one obvious bug, but the tests themselves appear to get jammed for spark and can't be run twice in a row.  Alright if I debug the rest of this in a different diff?

Comment (usmanm) [inline metis/metis/__init__.py]: Yeah, no worries. Might wanna do some bisection.

**Pull request merged**